src/main/java/KeyVaultClientEncryption.java

```diff
@@ -0,0 +1,223 @@
+        KeyClient keyClient = new KeyClientBuilder()
+                .vaultUrl(keyVaultUrl)
+                .credential(new DefaultAzureCredentialBuilder().build())
+                .buildClient();
+        // Get key for decryption using keyClient
+        KeyVaultKey key = keyClient.getKey(keyName);
+        String keyId = key.getId();
+        String command = "az storage account encryption-scope create --name " + encryptionScope + " -s " +
+                "Microsoft.KeyVault -u " + keyId +
+                " --account-name " + storageAccount + " -g " + resourceGroup + " --subscription " + subscription;
+
+        Process p = null;
+        try{
+            p = new ProcessBuilder("cmd.exe", "/c", command).start();
+        }
+        catch (IOException ex){
+            ex.printStackTrace();
+        }
+        BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()));
+        String line = null;
+        while (true) {
+            try {
+                line = r.readLine();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            if (line == null) { break; }
+//            System.out.println(line);
+    }
+
+    }
+
+    /**
+     * Creates example container and blob, then uploads with client-side encryption with key vault
+     **/
+    public static BlobClient setup(String storageAccount, String sharedKeyCred, String keyVaultUrl, String containerName, String blobName, String blobSuffix, String keyName){
+        String storageAccountUrl = "https://" + storageAccount + ".blob.core.windows.net";
+
+        // Creating a BlobServiceClient that allows us to perform container and blob operations, given our storage
+        // account URL and shared key credential
+        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .buildClient();
+
+        // Creating client referencing to-be-created container, and then creating it
+        BlobContainerClient blobContainerClient = blobServiceClient.getBlobContainerClient(containerName);
+        blobContainerClient.create();
+
+        // Creating key client that allows access of key vault
+        KeyClient keyClient = new KeyClientBuilder()
+                .vaultUrl(keyVaultUrl)
+                .credential(new DefaultAzureCredentialBuilder().build())
+                .buildClient();
+        // Creating an example RSA key in key vault
+        KeyVaultKey rsaKey = keyClient.createRsaKey(new CreateRsaKeyOptions(keyName)
+                .setExpiresOn(OffsetDateTime.now().plusYears(1))
+                .setKeySize(2048));
+        // Creating cryptography client that allows cryptographic operations using the key that was just created
+        CryptographyClient cryptoClient = new CryptographyClientBuilder()
```

Comment:
Here you should be using the AsyncKeyEncryptionKeyClient to make your key that points to keyvault and use the encryptedBlobClient like in the local case